src/test/java/com/example/onlinebookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,218 @@
+
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Verify the correct bookDto returns after book was saved")
+    public void saveBook_Valid_ShouldReturnBookDto() {
+        Long categoryId = 1L;
+        CreateBookRequestDto bookRequestDto = new CreateBookRequestDto(
+                "How to cook",
+                "Senior Tomato",
+                "123456789",
+                BigDecimal.valueOf(99),
+                "Very nice book",
+                "image src",
+                Set.of(categoryId)
+        );
+
+        Category category = new Category();
+        category.setId(categoryId);
+        category.setName("Adults");
+
+        Book book = new Book();
+        book.setTitle(bookRequestDto.title());
+        book.setAuthor(bookRequestDto.author());
+        book.setIsbn(bookRequestDto.isbn());
+        book.setPrice(bookRequestDto.price());
+        book.setDescription(bookRequestDto.description());
+        book.setCoverImage(bookRequestDto.coverImage());
+        book.setCategories(Set.of(category));
+
+        Mockito.when(categoryRepository.findById(categoryId)).thenReturn(Optional.of(category));
+        Mockito.when(bookRepository.save(Mockito.any(Book.class))).thenReturn(book);
+
+        BookDto expected = new BookDto();
+        expected.setTitle(bookRequestDto.title());
+        expected.setAuthor(bookRequestDto.author());
+        expected.setIsbn(bookRequestDto.isbn());
+        expected.setPrice(bookRequestDto.price());
+        expected.setDescription(bookRequestDto.description());
+        expected.setCoverImage(bookRequestDto.coverImage());
+        expected.setCategoriesIds(Set.of(category.getId()));
+
+        BookDto actual = bookService.save(bookRequestDto);
+
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
+    }
+
+    @Test
+    @DisplayName("Verify the correct exception message returns with not valid category id")
+    public void saveBook_NotValidCategoryId_ShouldReturnException() {
+        Long categoryId = -1L;
+        CreateBookRequestDto bookRequestDto = new CreateBookRequestDto(
+                "How to cook",
+                "Senior Tomato",
+                "123456789",
+                BigDecimal.valueOf(99),
+                "Very nice book",
+                "image src",
+                Set.of(categoryId)
+        );
+
+        Mockito.when(categoryRepository.findById(categoryId))
+                .thenThrow(EntityNotFoundException.class);
+
+        Assertions.assertThrows(EntityNotFoundException.class,
+                () -> bookService.save(bookRequestDto));
+    }
+
+    @Test
+    @DisplayName("Verify correct delete method invocation")
+    public void deleteBookById_ValidId_Success() {
+        Long id = 1L;
+
+        Mockito.doNothing().when(bookRepository).deleteById(id);
+
+        bookService.deleteById(id);
+
+        Mockito.verify(bookRepository, times(1)).deleteById(id);
```

Comment:
```suggestion
        Mockito.verify(bookRepository).deleteById(id);
```
times(1) is a default value, so you can skip it